cpp/density/LocalDensity.cc

```diff
@@ -41,56 +41,38 @@ void LocalDensity::compute(const box::Box& box, const freud::locality::NeighborL
     float area = M_PI * m_rcut * m_rcut;
     float volume = 4.0f/3.0f * M_PI * m_rcut * m_rcut * m_rcut;
     // compute the local density
-    freud::locality::forLoopWrapper(0, n_ref, 
-      [=] (size_t r_begin, size_t r_end)
-      {
-      size_t bond(nlist->find_first_index(r_begin));
-
-      for(size_t i=r_begin; i != r_end; ++i)
-          {
-          float num_neighbors = 0;
-
-          const vec3<float> r_i(ref_points[i]);
-
-          for(; bond < nlist->getNumBonds() && neighbor_list[2*bond] == i; ++bond)
-          {
-              const unsigned int j(neighbor_list[2*bond + 1]);
-
-              const vec3<float> r_j(points[j]);
-              const vec3<float> r_ij(m_box.wrap(r_j - r_i));
-
-              float rsq = dot(r_ij, r_ij);
-              float r = sqrt(rsq);
-
-              // count particles that are fully in the rcut sphere
-              if (r < (m_rcut - m_diameter/2.0f))
-              {
-                  num_neighbors += 1.0f;
-              }
-              else
-              {
-                  // partially count particles that intersect the rcut sphere
-                  // this is not particularly accurate for a single particle, but works well on average for
-                  // lots of them. It smooths out the neighbor count distributions and avoids noisy spikes
-                  // that obscure data
-                  num_neighbors += 1.0f + (m_rcut - (r + m_diameter/2.0f)) / m_diameter;
-              }
-          }
-
-          m_num_neighbors_array.get()[i] = num_neighbors;
-          if (m_box.is2D())
+    freud::locality::loopOverNeighborsIterator(ref_points, points, Np, qargs, nlist, 
```

Comment:
While this looks right, I'm making a note here that we need to thoroughly review the ref_points/points ordering since it is not quite consistent everywhere yet.